packages/components/src/components/text/text.jsx

```diff
@@ -4,25 +4,20 @@ import PropTypes from 'prop-types';
 import { isEmptyObject } from '@deriv/shared';
 
 const Text = ({ children, size, color, align, weight, line_height, as, className, styles, ...props }) => {
-    const setStyle = () => {
+    const [style, setStyle] = React.useState({});
+    React.useEffect(() => {
         const class_styles = {
             '--text-size': `var(--text-size-${size || 's'})`,
             '--text-color': `var(--text-${color || 'general'})`,
             '--text-lh': `var(--text-lh-${line_height || 'm'})`,
             '--text-weight': `var(--text-weight-${weight || 'normal'})`,
         };
         if (!isEmptyObject(styles)) {
-            return { ...class_styles, ...styles };
+            setStyle(...class_styles, ...styles);
         }
-        return class_styles;
-    };
+        setStyle(class_styles);
+    }, [size, color, align, weight, line_height]);
```

Comment:
Don't we need to call this effect just on mounting phase?

Reply:
Since, it is possible to send a prop dynamically after on mounting, I think we need to apply changes on updating as well.